packages/salesforcedx-utils-vscode/test/unit/commands/commandletExecutor.test.ts

```diff
@@ -102,6 +102,21 @@ describe('LibraryCommandletExecutor', () => {
     expect(showErrStub.called).to.be.true;
   });
 
+  it('should not show successful or failed notifications if run was cancelled', async () => {
+    const showErrStub = sb
+      .stub(vscodeStub.window, 'showErrorMessage')
+      .resolves(nls.localize('notification_unsuccessful_execution_text'));
+    const showInfoStub = sb
+      .stub(vscodeStub.window, 'showInformationMessage')
+      .resolves(nls.localize('notification_show_in_status_bar_button_text'));
+    const cancelledExecutor = new TestExecutor(new MockChannel());
+    cancelledExecutor.cancelled = true;
+
+    await cancelledExecutor.execute({ data: { success: true }, type: 'CONTINUE' });
+    expect(showErrStub.notCalled).to.be.true;
+    expect(showInfoStub.notCalled).to.be.true;
+  });
```

Comment:
Yeah it's a tricky test to write. Could probably do something like create our own cancellationTokenSource and make the vscodeStub.withProgress function call the task with the token from that source. But then we need to handle the event loop to make sure the onCancellationRequest listener is called before the test is over. I can try to do something like that in a subsequent PR, but this is a good start for some sanity.

Reply:
yeah, we can mock withProgress to be like:
```
withProgressStub.callsFake((options, task) => {
   // create our own mock token
   // See the (over)simplified version of cancellation token in the design doc
    task(mockProgress, mockToken);
});
// we can cancel because we own the token
```

Reply:
I tried this approach and it didn't behave as expected. I don't want to block bryan and shelby's story so I've left it as is for now and I'll address this and remove the cancellable class property in a follow up story

Reply:
Sounds good. We can address that later which is more about internal engineering. The class property is private and doesn't affect this API